pkg/workflows/upgrade.go

```diff
@@ -535,6 +533,13 @@ func (s *upgradeWorkloadClusterTask) Run(ctx context.Context, commandContext *ta
 		return &CollectDiagnosticsTask{}
 	}
 
+	logger.V(4).Info("Updating machine health checks on cluster")
+	err = commandContext.ClusterManager.UpdateMachineHealthChecks(ctx, commandContext.ClusterSpec, commandContext.ManagementCluster)
```

Comment:
could we let the controller do this for us?

Reply:
what about reconciling management cluster's mhcs tho?